app/controllers/sentences_controller.php

```diff
@@ -773,6 +776,35 @@ public function search()
             $tags = implode(',', $tagsArray);
         }
 
+        // filter by list
+        $searchableLists = $this->SentencesList->getSearchableLists();
+        if (!empty($list)) {
+            $isSearchable = $this->SentencesList->isSearchableList($list);
+            if ($isSearchable) {
+                $sphinx['filter'][] = array('lists_id', $list);
+                $found = false;
+                foreach ($searchableLists as $rec) {
+                    if ($list == $rec['SentencesList']['id']) {
+                        $found = true;
+                        break;
+                    }
+                }
+                if (!$found) {
+                    $searchableLists[] = $isSearchable;
+                }
+            } else {
+                $ignored[] = format(
+                    /* @translators: This string will be preceded by
+                       “Warning: the following criteria have been
+                       ignored:” */
+                    __("“belongs to list number {listId}”, because this list does ".
```

Comment:
By “cannot be searched” I mean that we prevent the user from using that list as a search criterion. The reason is that the list is private (more precisely, it is private while not owned by the user), but it may change in the future (or not; maybe I’m just over-abstracting things).

That said, when a member hands over a search link filtering on a private list to someone, a warning message talking about private list would be way more useful, as it gives him a hint about the need to switch the privacy setting of the list.